dataframe-csv/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/io/DataFrameCustomDoubleParser.kt

```diff
@@ -0,0 +1,24 @@
+package org.jetbrains.kotlinx.dataframe.impl.io
+
+import io.deephaven.csv.containers.ByteSlice
+import io.deephaven.csv.tokenization.Tokenizer.CustomDoubleParser
+import org.jetbrains.kotlinx.dataframe.api.ParserOptions
+
+/**
+ * Wrapper around [FastDoubleParser] so we can use it from Deephaven.
+ */
+internal class DataFrameCustomDoubleParser(parserOptions: ParserOptions) : CustomDoubleParser {
+
+    private val fastDoubleParser = FastDoubleParser(parserOptions)
+
+    override fun parse(bs: ByteSlice): Double =
+        try {
+            fastDoubleParser.parseOrNull(bs.data(), bs.begin(), bs.size())
+        } catch (e: Exception) {
+            null
+        } ?: throw NumberFormatException("Failed to parse double")
+
+    override fun parse(cs: CharSequence): Double =
+        fastDoubleParser.parseOrNull(cs.toString())
+            ?: throw NumberFormatException("Failed to parse double")
```

Comment:
maybe it will be useful to include substring to exception message

Reply:
the message is already debug-logged in parseOrNull.
This `NumberFormatException` is just for Deephaven to catch. Actually, I can even remove the message altogether, as Deephaven doesn't do anything with it either. (they just catch it and return false)